src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java

```diff
@@ -60,6 +61,24 @@ public void findEnvelopesToResend_should_filter_based_on_status_and_jurisdiction
         assertThat(result).hasSize(1);
     }
 
+    @Test
+    public void findByZipFileName_should_find_envelops_in_db() {
+        // given
+        dbHas(
+            envelope("A.zip", "X", Status.PROCESSED),
+            envelope("A.zip", "Y", Status.UPLOAD_FAILURE),
+            envelope("B.zip", "Z", Status.UPLOAD_FAILURE)
+        );
+
+        // when
+        List<Envelope> resultForA = repo.findByZipFileName("A.zip");
+        List<Envelope> resultForX = repo.findByZipFileName("X.zip");
+
+        // then
+        assertThat(resultForA).hasSize(2);
```

Comment:
Maybe we could do a similar thing to what's in ProcessEventRepositoryTest, i.e. `containsExactly...`?

Reply:
@lgonczar I've followed other tests in this class. There is one test that checks mapping 
Note that this new repo method does not include a custom SQL query, jpa magically builds a query based on the method name. So just checking the filtering part here.

Reply:
@lgonczar we can't - we are fully aware of that. we can write for conceptual matter, but what you are referring (`containsExactly...`) is addressed in unit test. no need to double check type safe ecosystem. integration test is more of a "does this definitely return a list at all times" given the unit test covers the rest